### PR TITLE
Allow callable in dehydrate_method

### PR DESCRIPTION
**Pull request by TheRealVizard**
**Problem**

Can't reuse a `dehydrate` function across multiple Fields

**Solution**

Allow a callable to be set as the `dehydrate_method`

**Acceptance Criteria**

Run several local tests.

### CODE REVIEW

**Review by matthewhegarty — COMMENTED** [review]
Looks like a good addition, thanks.

Should we add a small note to the documentation here:

https://django-import-export.readthedocs.io/en/latest/advanced_usage.html#advanced-data-manipulation-on-export

File: tests/core/tests/test_fields.py [hunk 1]
```diff
@@ -110,6 +110,14 @@ def testget_dehydrate_method_with_custom_method_name(self):
         method_name = field.get_dehydrate_method(resource_field_name)
         self.assertEqual(method_name, custom_dehydrate_method)
 
+    def testget_dehydrate_method_with_callable(self):
```
Comment (matthewhegarty) [inline tests/core/tests/test_fields.py:113]: should the test be named `test_*` for consistency?

Comment (TheRealVizard) [reply tests/core/tests/test_fields.py]: yes my bad fixed in [60de1ef](https://github.com/django-import-export/django-import-export/pull/1950/commits/60de1efd887853ffc02b7c7b79c7ca2e44a7991f)

**Comment by TheRealVizard** [comment]
> Looks like a good addition, thanks.
> 
> Should we add a small note to the documentation here:
> 
> https://django-import-export.readthedocs.io/en/latest/advanced_usage.html#advanced-data-manipulation-on-export

Added in [33a06e0](https://github.com/django-import-export/django-import-export/pull/1950/commits/33a06e01ecdabfdf0e90d80eb1234bc458621728) and fixed example in [4dd6584](https://github.com/django-import-export/django-import-export/pull/1950/commits/4dd6584834783c9431d091742510ae472c97311f)

File: docs/advanced_usage.rst [hunk 2]
```diff
@@ -759,9 +759,30 @@ In this case, the export looks like this:
...
+            model = Book
+
+        def custom_dehydrate_method(self, book):
+            return f"{book.name} by {book.author.name}"
+
+    # Using a callable directly
+    def custom_dehydrate_callable(book):
+        return f"{book.name} by {book.author.name}"
+
+    class BookResource(resources.ModelResource):
```
Comment (matthewhegarty) [inline docs/advanced_usage.rst:780]: Looks like you have some duplication here

**Review by matthewhegarty — CHANGES_REQUESTED** [review]
What is the benefit of defining a callable vs using the existing `dehydrate_<field>()`?  Both take the instance as a param so is there a specific use-case for using a callable?

**Comment by TheRealVizard** [comment]
> What is the benefit of defining a callable vs using the existing `dehydrate_<field>()`? Both take the instance as a param so is there a specific use-case for using a callable?

Allowing the use of callables makes it easier to apply a single dehydrate function across multiple Resource classes that cannot inherit from each other or across different fields. This provides greater flexibility.

Additionally, this change allows the use of lambdas for simple transformations or calculations, making it even more convenient to apply quick, inline modifications without the need to define separate methods.

File: docs/advanced_usage.rst (hunk 2 above)
Comment (TheRealVizard) [reply docs/advanced_usage.rst:780]: This is to illustrate how use both using the `dehydrate_method` with a callable and with a string

**Review by matthewhegarty — APPROVED** [review]
Thanks for submitting

**Review by matthewhegarty — APPROVED** [review]
Thanks for submitting

**Pull request merged**